### ISSUE TITLE
Cherry pick bug fixes from Greenplum to CBDB

### ISSUE THREAD

**Issue by my-ship-it** [cloudberrydb/cloudberrydb#133]

**Comment by lss602726449** [comment]
We use the merge-base to get the common ancestor
```
git merge-base main gpdb/main
d5573b984b37cccbe74078cdf986017771d09ba2
```
Here are some log in gpdb,and I will cherry-pick some bug and intersting enhancement from the gpdb
```
...
10296094c9900a6a9161ddb71e8184ca6f7aa45a removed redefined PG_AUTOCONF_FILENAME in guc.h (#13081)
f13f7dd6d4ef36948eb7a05142ab475ed3dd49d3 Docs - fix the broken link to a non-existing doc (#13078)
61c20a5eaac3a77ff52a3a47ceb64ff238192281 Make pg_ctl stop/restart/promote recheck postmaster aliveness.
64dd648072958b7e91e2ea289fcb106acae3bfba Use gendef instead of pexports for building windows .def files
29a9f92fcbb3fd21ddbd7fe0bb61c9abb0d28457 Fix getResUsage integer overflow.
3a32e53e1f383090f27b23d4356eb1d19172ef1b Use Test::Builder::todo_start(), replacing $::TODO.
2d96a66c2eef76f44cc9f9eca1a611330e215dc0 Fix back-patch of "Avoid race in RelationBuildDesc() ..."
d5573b984b37cccbe74078cdf986017771d09ba2 docs - clarify the use and value of diskquota.max_active_tables cfg p… (#13059)
```

**Comment by my-ship-it** [comment]
Already in progress.
